src/zebra_puzzles/evaluation.py

```diff
@@ -223,30 +233,90 @@ def query_llm(prompt: str, model: str, response_format: Type[BaseModel]) -> Base
             temperature=0,
             seed=42,
             response_format=response_format,
+            max_completion_tokens=100_000,
+            reasoning_effort="medium",
         )
     except BadRequestError as e:
-        if "'temperature' is not supported" in str(e):
-            response = client.beta.chat.completions.parse(
-                messages=[{"role": "user", "content": prompt}],
-                model=model,
-                seed=42,
-                response_format=response_format,
-            )
-        else:
-            raise e
+        max_retries = 5
+        wait_time = 5
+        # gpt-4o-mini
+        if "'Unrecognized request argument supplied: reasoning_effort'" in str(e):
+            while retries := 0 < max_retries:
```

Comment:
Nit: Since this is a loop from 0 to 5, this could just be a for loop 🙂 

You could look into for-else loops, they're quite handy for these kinds of retry-loops.